lib/pluginmanager/package.rb

```diff
@@ -0,0 +1,67 @@
+class LogStash::PluginManager::Package < LogStash::PluginManager::Command
+
+  option "--tgz", :flag, "compress package as a tar.gz file", :default => false
+  option "--zip", :flag, "compress package as a zip file", :default => false
+  option "--clean", :flag, "clean up the generated dump of plugins", :default => false
+  option "--override", :flag, "Override a previously generated package file", :default => false
+
+  def execute
+    signal_error("File #{LogStash::Environment::GEMFILE_PATH} does not exist or is not writable, aborting") unless File.writable?(LogStash::Environment::GEMFILE_PATH)
+    puts("packaging plugins for offline usage")
+    if override?
+      rm(target_file)
+    else
+      validate_target_file
+    end
+    LogStash::Bundler.invoke!({:package => true, :all => true})
+    compress_cache(LogStash::Environment::CACHE_PATH)
+    rm_rf(LogStash::Environment::CACHE_PATH) if clean?
+  end
+
+  private
+
+  def compress_cache(location)
+    require "logstash/util/compress"
+    if zip?
+      LogStash::Util::Zip.compress(location, target_file)
+    elsif tgz?
+      LogStash::Util::Tar.compress(location, target_file)
+    end
+  end
+
+  def validate_target_file
+    if File.exist?(target_file)
+      puts("File #{target_file} exist, do you want to override it to continue? (Y/N)")
+      override = ( "y" == STDIN.gets.strip.downcase ? true : false)
+      if override
+        File.delete(target_file)
+      else
+        puts("To continue you need to save the #{target_file} to a save place and redo the process")
+        exit
+      end
+    end
+  end
+
+  def target_file
+    target_file = File.join(LogStash::Environment::LOGSTASH_HOME, "plugins_cache")
```

Comment:
We could probably move this path in the `Logstash::Environment` class? It will be easier to reuse if needed.